### PR TITLE
Outreachy week nine/update android tests

### PR DESCRIPTION
**Pull request by Lamouresparus**
This PR:
- Replace deprecated InstrumentationRegistry.getContext() with ApplicationProvider.getApplicationContext()
- Handle null pointer exception 
- Simplify assertions
- Clean up code

### CODE REVIEW

**Comment by wbrunette** [comment]
runtests

File: services_app/src/androidTest/java/org/opendatakit/database/service/GroupsPermissionTest.java [hunk 1]
```diff
@@ -601,7 +602,7 @@ public void testUserAuthorizationFailure() {
          assertEquals(1, table.getNumberOfRows());
          row = table.getRowAtIndex(0);
 
-         assertEquals(2L,row.getDataByKey(COL_INTEGER_ID));
+         assertEquals(Long.valueOf(2),row.getDataByKey(COL_INTEGER_ID));
```
Comment (wbrunette) [inline services_app/src/androidTest/java/org/opendatakit/database/service/GroupsPermissionTest.java:605]: This might make more sense being a constant? What do you think? So it's clear in the test when the value is being populated and when the value is being checked they are the same?

Comment (Lamouresparus) [reply services_app/src/androidTest/java/org/opendatakit/database/service/GroupsPermissionTest.java:605]: Yes, I do think so too. I have created constants COL_INTEGER_VALUE_TWO, COL_INTEGER_VALUE_THREE, and COL_INTEGER_VALUE_FOUR that stores int 2, 3, and 4 respectively. This way, we can reuse the values and quickly identify its usages.

File: services_app/src/androidTest/java/org/opendatakit/database/service/GroupsPermissionTest.java [hunk 2]
```diff
@@ -858,7 +861,7 @@ public void testGroupAuthorizationAnonymousFailureLocked_ExpectSomeFailures() {
          assertEquals(1, table.getNumberOfRows());
          row = table.getRowAtIndex(0);
 
-         assertEquals(Long.valueOf(2), row.getDataByKey(COL_INTEGER_ID));
+         assertEquals(Long.valueOf(COL_INTEGER_VALUE_TWO), row.getDataByKey(COL_INTEGER_ID));
```
Comment (wbrunette) [inline services_app/src/androidTest/java/org/opendatakit/database/service/GroupsPermissionTest.java:864]: Why not make the constant an Object Long?

Comment (Lamouresparus) [reply services_app/src/androidTest/java/org/opendatakit/database/service/GroupsPermissionTest.java:864]: Hmmmm. Okay then. I used an int because we were initially populating the contetnt values with them.

File: services_app/src/androidTest/java/org/opendatakit/database/service/OdkDatabaseServiceImplTest.java [hunk 3]
```diff
@@ -188,7 +190,8 @@ protected void tearDownBefore() {
          for (Row row : rows) {
             assertEquals(row.getRawStringByKey(COLUMN_ID1), "ayy lmao");
```
Comment (wbrunette) [inline services_app/src/androidTest/java/org/opendatakit/database/service/OdkDatabaseServiceImplTest.java:191]: Make "ayy lmao" and "3" constants

File: services_app/src/androidTest/java/org/opendatakit/database/service/OdkDatabaseServiceTest.java [hunk 4]
```diff
@@ -308,7 +312,8 @@ public void testDbInsertThrowsIllegalArgumentException()
          List<Column> columnList = createColumnList();
          ColumnList colList = new ColumnList(columnList);
 
-         db = serviceInterface.openDatabase(APPNAME);
+          assertNotNull(serviceInterface);
```
Comment (wbrunette) [inline services_app/src/androidTest/java/org/opendatakit/database/service/OdkDatabaseServiceTest.java:315]: why did you add an assertNotNull for serviceInterface but not for DBHandle? Seem like we should be consistent. Maybe it's impossible for DbHandle to be null (you might want to check on that)

**Review by wbrunette — APPROVED** [review]
